src/base_alloc/base_alloc_linear.c

```diff
@@ -69,26 +105,59 @@ umf_ba_linear_pool_t *umf_ba_linear_create(size_t pool_size) {
 
 void *umf_ba_linear_alloc(umf_ba_linear_pool_t *pool, size_t size) {
     size_t aligned_size = align_size(size, MEMORY_ALIGNMENT);
-
     util_mutex_lock(&pool->metadata.lock);
     if (pool->metadata.size_left < aligned_size) {
-        fprintf(stderr,
-                "error: umf_ba_linear_alloc() failed (requested size: %zu > "
-                "space left: %zu)\n",
-                aligned_size, pool->metadata.size_left);
-        util_mutex_unlock(&pool->metadata.lock);
-        return NULL; // out of memory
+        // assert in Debug build (OOM)
+        assert(pool->metadata.size_left >= aligned_size);
```

Comment:
Why are we failing in Debug and succeeding in Release? Why can't we have a consistent behavior?